static/js/compose_actions.js

```diff
@@ -384,7 +384,13 @@ exports.quote_and_reply = function (opts) {
     });
 };
 
-exports.on_narrow = function () {
+exports.on_narrow = function (opts) {
+    // If force_close is an option do not open compose box to better use navigation
+    if (opts.force_close) {
+        exports.cancel();
+        return;
+    }
```

Comment:
Does this just need to be `return`?  I'm not sure whether the `cancel` is what's expected here.

Reply:
Those are the semantics I expect--to actually force-close the compose box.

Reply:
@timabbott This is to handle the case, if the compose box was already open in the other narrow and 'p' hotkey is pressed. It then closes the compose box and switches to the unread PM. @showell 's review above mentions this.
Without the `cancel` the compose box would be open and would be focus while changing to the unread PM.

Reply:
OK, yeah, I think that's correct.  Thanks for the explanation!

Can you add a comment next to where we pass `force_close` explaining what it's for?  Just something like "When using the `p` hotkey to jump through PM threads, one wants to be able to do it more than once in a row, which requires compose not popping open"

Reply:
@timabbott [Here](https://github.com/zulip/zulip/pull/8409/files#diff-e1eab9dc9b92b9773c2295bcb5f60b9fR382) is a comment explaining the use of `force_close` in general.